src/Functions/DateTimeTransforms.h

```diff
@@ -55,7 +55,7 @@ constexpr time_t MAX_DATETIME64_TIMESTAMP = 10413791999LL;    //  1900-01-01 00:
 constexpr time_t MIN_DATETIME64_TIMESTAMP = -2208988800LL;    //  2299-12-31 23:59:59 UTC
 constexpr time_t MAX_DATETIME_TIMESTAMP = 0xFFFFFFFF;
 constexpr time_t MAX_DATE_TIMESTAMP = 5662310399;       // 2149-06-06 23:59:59 UTC
-constexpr time_t MAX_DATETIME_DAY_NUM =  49710;               // 2106-02-07
+constexpr time_t MAX_DATETIME_DAY_NUM =  49709;         // 2106-02-06 America/Hermosillo
```

Comment:
why we reduce it by one day?

Reply:
Because:

```
SELECT toDateTime(toDate(49710), 'America/Hermosillo')

   ┌─toDateTime(toDate(49710), 'America/Hermosillo')─┐
1. │                             1969-12-31 16:31:44 │
   └─────────────────────────────────────────────────┘

SELECT toDateTime(toDate(49710), 'UTC')

   ┌─toDateTime(toDate(49710), 'UTC')─┐
1. │              2106-02-07 00:00:00 │
   └──────────────────────────────────┘
```